src/main/scala/dxWDL/compiler/Tree.scala

```diff
@@ -57,35 +56,111 @@ case class Tree(execDict: Map[String, ExecRecord]) {
     }
   }
 
-  def prettyPrint(primary: Native.ExecRecord, indent: Int = 0): String = {
-    primary.callable match {
-      case apl: IR.Applet if apl.kind.isInstanceOf[AppletKindTask] =>
-        Utils.genNSpaces(indent) + Console.GREEN + apl.name + Console.RESET
+  /** Recursivly traverse the exec tree and generate an appropriate name + color based on the node type.
+    * The prefix is built up as recursive calls happen. This allows for mainaining the locations of branches
+    * in the tree. When a prefix made for a current node, it undergoes a transformation to strip out any
+    * extra characters from previous calls. This maintains the indenation level and tree branches.
+    *
+    * Color scheme:
+    *   'types' are in CYAN, types being Workflow, App, Task etc
+    *   'names' are in WHITE for all app types, and YELLOW for workflow types
+    *
+    * Step by step:
+    *
+    *       TREE                                            LEVEL               prettyPrint calls (approx)                                                         NOTES
```

Comment:
View in file, not in commit, the lines went long fitting in the line by line step through.